src/realm/object-store/sync/sync_session.cpp

```diff
@@ -1061,7 +1082,9 @@ void SyncSession::resume()
 void SyncSession::do_revive(util::CheckedUniqueLock&& lock)
 {
     auto u = user();
-    if (!u || !u->access_token_refresh_required()) {
+    // If the sync manager doesn't have a valid route yet, then request
+    // an updated access token to refresh it.
+    if (m_sync_manager->sync_route() && (!u || !u->access_token_refresh_required())) {
```

Comment:
While correct, I found it confusing to read this comment in the negative while the code is phrased in the positive. It might be easier to read "If the sync manager has a valid route and the user's token appears valid try to start the session, otherwise request a new token to authenticate the user"